admin/services/data-access/pupil.data.service.js

```diff
@@ -152,6 +152,36 @@ pupilDataService.sqlFindOneByUpn = async (upn = '') => {
   return R.head(results)
 }
 
+/**
+ * Find a pupil by upn within school
+ * @param {String} upn
+ * @param {Number} schoolId
+ * @return {Promise<void>}
+ */
+pupilDataService.sqlFindOneByUpnAndSchoolId = async (upn = '', schoolId) => {
```

Comment:
why the empty string default, is that a possible case?

Reply:
@GuyHarwood I made a change where lookup should be prevented if the upn that is passed down has already validation errors.
The data service is now modified as well with no default as a value expectation